examples/teleport-usage/teleport_usage.py

```diff
@@ -0,0 +1,123 @@
+if not start_date:
+    now = datetime.datetime.now()
+    start_date = now - datetime.timedelta(days=30)
+else:
+    start_date = datetime.datetime.strptime(start_date,'%Y-%m-%d')    
+
+end_date = start_date + datetime.timedelta(days=30)
+
+# Configure DynamoDB for query
+dynamodb = boto3.resource("dynamodb", region_name=aws_region)
+table = dynamodb.Table(TABLE_NAME)
+
+# variables for protocols
+app = 'app.session.start'
+db = 'db.session.start'
+kube = 'kube.request'
+ssh = 'session.start'
+win = 'windows.desktop.session.start'
+
+protocols = ['app', 'db', 'kube', 'win']
+all_users = []
+
+# Generating data message
+start_date_str = start_date.strftime("%b %d %y")
+end_date_str = end_date.strftime("%b %d %y")
+day_range = (end_date - start_date).days
+d_range = (end_date - start_date).days + 1
+print("""
+NOTE: By default this tool queries usage for the last 30 days") 
+For a custom date range use ENV START_DATE and END_DATE (format YYYY-MM-DD)")
+""")
+print(f"Gathering data for unique users who accessed each protocol between {start_date_str} to {end_date_str} ({day_range} days): ")
+
+unique_users = set()
+
+# Run Dynamodb query to retrieve all login for SSH for specified date range
+for i in range(d_range):
+    current_date = start_date + datetime.timedelta(days=i)
+
+    date = current_date.strftime("%Y-%m-%d")
+    response = table.query(
+        TableName=TABLE_NAME,
+        IndexName="timesearchV2",
+        KeyConditionExpression=Key("CreatedAtDate").eq(date),
+        FilterExpression=Attr("EventType").begins_with(ssh),
+    )
+    # Calculate unique users
+    for item in response["Items"]:
+        user = item.get("FieldsMap").get("user")
+        if user:
+            unique_users.add(user)
+print(f"ssh: {len(unique_users)}")
+if len(unique_users) != 0:
+    all_users.append(f"ssh: {unique_users}")
+
+# Loop through the protocols in the list
+for protocol in protocols:
+  # Create a set for the unique users of this protocol
+  unique_users = set()
+
+  # Run Dynamodb query to retrieve all login event types for specified date range
+  for i in range(d_range):
+      current_date = start_date + datetime.timedelta(i)
+      date = current_date.strftime("%Y-%m-%d")
+      response = table.query(
+          TableName=TABLE_NAME,
+          IndexName="timesearchV2",
+          KeyConditionExpression=Key("CreatedAtDate").eq(date),
+          FilterExpression=Attr("EventType").begins_with(protocol),
+      )
+      # Calculate unique users
+      for item in response["Items"]:
+          user = item.get("FieldsMap").get("user")
+          if user:
+              unique_users.add(user)
+
+  # Print the count of unique users for this protocol
+  print(f"{protocol}: {len(unique_users)}")
+  if len(unique_users) != 0:
+      all_users.append(f"{protocol}: {unique_users}")
+
+# By default will not print all users, to print use ENV  SHOW_USERS=YES
+print_users = os.getenv("SHOW_USERS")
+if print_users == "YES":

```

Comment:
Any reason not to print this all the time?

Reply:
We opted for default behavior to not printing the names unless they specified the flag to reduce the noise of the output.